Classes/ViewHelpers/GetCountImgViewHelper.php

```diff
@@ -14,33 +14,54 @@
  * (c) 2024 Christoph Daecke <typo3@mediadreams.org>
  */
 
-use TYPO3\CMS\Core\Utility\GeneralUtility;
-use TYPO3Fluid\Fluid\Core\Rendering\RenderingContextInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use TYPO3\CMS\Core\Information\Typo3Version;
 use TYPO3Fluid\Fluid\Core\ViewHelper\AbstractViewHelper;
-use TYPO3Fluid\Fluid\Core\ViewHelper\Traits\CompileWithRenderStatic;
 
 final class GetCountImgViewHelper extends AbstractViewHelper
 {
-    use CompileWithRenderStatic;
     protected $escapeOutput = false;
 
     public function initializeArguments(): void
     {
         $this->registerArgument('newsUid', 'int', 'The uid of the news record', true);
-        $this->registerArgument('pageUid', 'int', 'The uid of the page', true);
+        $this->registerArgument('pageUid', 'int', 'The uid of the page');
     }
 
-    public static function renderStatic(
-        array $arguments,
-        \Closure $renderChildrenClosure,
-        RenderingContextInterface $renderingContext
-    ): string {
-        $siteUrl = $renderingContext->getRequest()->getAttribute('normalizedParams')->getSiteUrl();
-
-        return sprintf(
-            '<img src="%smd-newsimg-%s.gif" alt="" width="1" height="1" aria-hidden="true" />',
-            $siteUrl,
-            $arguments['newsUid']
-        );
+    public function render(): string
+    {
+        // TODO: $this->arguments['pageUid'] is obsolet and can be removed in v5.0
+        if (isset($this->arguments['pageUid'])) {
+            trigger_error(
+                'Parameter "pageUid" in "GetCountImgViewHelper" is deprecated since version 4.0.0, will be removed in version 5.0.0',
+                E_USER_DEPRECATED
+            );
+        }
+
+        $request = $this->getRequest();
+
+        if ($request instanceof ServerRequestInterface) {
```

Comment:
Please use early return like you had it in the commit before

Reply:
Hmm, in your previous comment, you suggested to remove the `else`. How can I return early AND remove the else? Could you please you show your suggestion by providing a code sample?

Reply:
```php
        if (!$request instanceof ServerRequestInterface) {
            return '';
        }

        $siteUrl = $request->getAttribute('normalizedParams')->getSiteUrl();

        return sprintf(
            '<img src="%smd-newsimg-%s.gif" alt="" width="1" height="1" aria-hidden="true" />',
            $siteUrl,
            $this->arguments['newsUid']
        );
```

Reply:
Sometimes it needs some time... Now I got it ;-)